NeoMathEngine/src/CPU/x86/NeoMathEngineAvxDll.cpp

```diff
@@ -0,0 +1,109 @@
+/* Copyright © 2017-2020 ABBYY Production LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#include <common.h>
+#pragma hdrstop
+
+#if FINE_PLATFORM( FINE_DARWIN ) || FINE_PLATFORM( FINE_LINUX ) || FINE_PLATFORM( FINE_ANDROID )
+#include <cpuid.h>
+#endif
+#if FINE_PLATFORM( FINE_WINDOWS )
+#include <intrin.h>
+#endif
+
+#include <NeoMathEngineAvxDll.h>
+#include <MathEngineCommon.h>
+#include <MemoryHandleInternal.h>
+
+namespace NeoML {
+
+CNeoMathEngineAvxDll::CNeoMathEngineAvxDll() : isLoaded( false ), functionAdresses{}
+{
+	if( !isAvxAvailable() || !Load( libName ) ) {
```

Comment:
It should be an error!

If( !isAvxAvailable() ) {
 return;
}
isLoaded = Load( libName );
assert( isLoaded );

Reply:
for cpu without AVX it should just return.